x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/PreAnalyzer.java

```diff
@@ -96,8 +99,16 @@ protected PreAnalysis doPreAnalyze(LogicalPlan plan) {
             if (fn.name().equalsIgnoreCase("to_aggregate_metric_double")) {
                 useAggregateMetricDoubleWhenNotSupported.set(true);
             }
+            if (List.of("to_date_range", "to_string").contains(fn.name().toLowerCase(Locale.ROOT))) {
```

Comment:
I don't understand the point of this code. It seems to relate to the comment in IndexResolver:279, which seems a copy of the aggregate-metric-double comment related to older cluster nodes supporting that type without reporting that they do. Is that really true for date_range too?